x-pack/plugins/data_enhanced/public/search/async_search_strategy.test.ts

```diff
@@ -96,6 +96,34 @@ describe('Async search strategy', () => {
       });
   });
 
+  // For bug fixed in https://github.com/elastic/kibana/pull/64155
+  it('Continues polling if no records are returned on first async request', async () => {
+    mockSearch
+      .mockReturnValueOnce(of({ id: 1, total: 0, loaded: 0, is_running: true, is_partial: true }))
+      .mockReturnValueOnce(
+        of({ id: 1, total: 2, loaded: 2, is_running: false, is_partial: false })
```

Comment:
Would this case indicate that the search is done? I would expect at least the `is_partial` flag to be `true` here. Or am I missing something?

Reply:
@alexh97 
indeed! 
The first response is the one that was problematic before the fix - we used to get 0 total, so we would stop polling.
Now we continue polling because is_running and is_partial are true.

This second response is just a normal completion.